### PR TITLE
Carousel View Implementation on Android

### PR DESCRIPTION
**Pull request by meethjain**
# Description

This PR introduces implementation of Carousel Component for Android

Screenshot 2024-10-15 at 7 37 44 PM

Reference Video

1. Carousel Component Sample 1 (Light Theme)
[Android_Carousel_1_Light.webm](https://github.com/user-attachments/assets/c6c8d882-d7ea-4375-b609-24e6efeaa6f6)

2. Carousel Component Sample 1 (Dark Theme)
[Android_Carousel_1_Dark.webm](https://github.com/user-attachments/assets/ac604978-8448-4a55-85ae-382fe492ff93)

3. Carousel Component Sample 2 (Light & Dark Theme)
[Android_Carousel_2.webm](https://github.com/user-attachments/assets/be4a7108-ddd9-4e34-95be-3aaf762d60f4)

# Local Verification on Teams

Teams Local Integration 
[Teams_Carousel.webm](https://github.com/user-attachments/assets/8e2aecaf-9367-46ae-9b14-330d755965a0)

Note: Adapto Bot is currently not responding for Carousel Component on Desktop, Android & iOS hence hardcoded the same)

## Next Set of Items
- Implementation of custom scrolling indicator for the carousel effect (currently using TabLayout)
- Addressing Height Change when elements with larger height is later

### CODE REVIEW

**Comment by meethjain** [comment]
@microsoft-github-policy-service agree company="Microso

**Comment by meethjain** [comment]
@microsoft-github-policy-service agree

**Review by prpatwa — APPROVED** [review]
Apart from view pager adapter comment, Other changes LGTM.

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/carousel/CarouselPageAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.carousel_item_viewpager, parent, false)
+        return CarouselPageHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: CarouselPageHolder, position: Int) =
+        holder.bind(position, pages[position], renderedCard, cardActionHandler, hostConfig, renderArgs, fragmentManager)
+
+    class CarouselPageHolder(view: View) : RecyclerView.ViewHolder(view) {
+
+        private val views = mutableMapOf<Int, View>()
```
[the remark this answers is not in the document]
Comment (meethjain) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/carousel/CarouselPageAdapter.kt:45]: Sure, taking this up immediately in the follow up PR

**Review by guptaabhishe — APPROVED** [review]

File: samples/v1.6/Elements/Carousel.sample.json [hunk 2]
```diff
@@ -35,7 +35,68 @@
                             "wrap": true,
                             "separator": true,
                             "spacing": "Medium"
-                        }
+                        },
+						{
```
[the remark this answers is not in the document]
Comment (meethjain) [reply samples/v1.6/Elements/Carousel.sample.json:39]: Fixed the indentation

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/carousel/CarouselPageAdapter.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (meethjain) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/carousel/CarouselPageAdapter.kt]: Verified & added the changes

**Review by guptaabhishe — APPROVED** [review]

**Comment by meethjain** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Pull request merged**